core/src/memory.rs

```diff
@@ -584,6 +588,7 @@ impl AllocationsContext {
         &mut self,
         pages: WasmPage,
         mem: &mut impl Memory,
+        charge_gas_for_grow: impl FnOnce(WasmPage) -> Result<(), ChargeError>,
     ) -> Result<AllocInfo, AllocError> {
```

Comment:
The only one reason to return AllocInfo instead of pure allocated page is to return gas. So this must be removed